Include/internal/pycore_jit.h

```diff
@@ -11,7 +11,7 @@ extern "C" {
 
 #ifdef _Py_JIT
 
-typedef _Py_CODEUNIT *(*jit_func)(_PyInterpreterFrame *frame, _PyStackRef *stack_pointer, PyThreadState *tstate);
```

Comment:
Let's leave this as `jit_func` (and the new one as `jit_func_preserve_none`). I think the native calling convention can be implied.